packages/grid/_modules_/grid/hooks/features/rows/useGridRows.ts

```diff
@@ -20,8 +20,8 @@ import { useLogger } from '../../utils/useLogger';
 import { useGridState } from '../core/useGridState';
 import { getInitialGridRowState, InternalGridRowsState } from './gridRowsState';
 
-export function addGridRowId(rowData: GridRowData, getRowId?: GridRowIdGetter): GridRowModel {
-  return getRowId == null ? (rowData as GridRowModel) : { id: getRowId(rowData), ...rowData };
+function getGridRowId(rowData: GridRowData, getRowId?: GridRowIdGetter): GridRowId {
+  return getRowId ? getRowId(rowData) : rowData.id;
```

Comment:
Should we call `checkGridRowIdIsValid` here?
only if `getGridRowId` is always followed by `checkGridRowIdIsValid`

Reply:
Changed to also call `checkGridRowIdIsValid` after computing the id.